src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Union
+
+from affine import Affine
+from cv2 import INTER_LINEAR, INTER_NEAREST, resize
+from numpy import array, full
+from numpy.typing import NDArray
+from rasterio.drivers import raster_driver_extensions
+from rioxarray.rioxarray import affine_to_coords
+from xarray import DataArray, Dataset, merge, open_dataset
+
+from sds_data_model._vector import _check_layer_projection
+from sds_data_model.constants import BNG_XMAX, BNG_XMIN, BNG_YMAX, CELL_SIZE
+
+
+def _has_wrong_cell_size(
+    transform: Affine,
+    expected_cell_size: int,
+) -> bool:
+    """Return true if the cell size of x (a) and y (e) in the \
+    input transform are **not equal** to the expected cell size.
+
+    Args:
+        transform (Affine): Input transform, the data being checked.
+        expected_cell_size (int): The expected cell size in the transform.
+
+    Returns:
+        bool: An indication in `_resample_and_reshape whether` to resample data.
+    """
+    if transform.a != expected_cell_size or -transform.e != expected_cell_size:
+        return True
+    else:
+        return False
+
+
+def _has_wrong_shape(
+    transform: Affine,
+    expected_x_min: int,
+    expected_y_max: int,
+) -> bool:
+    """Return true if the shape/extent of the input data \
+    is **not equal** to the expected shape.
```

Comment:
As above, the first line of a function docstring should should be a one line (<88 character) description of the function.

```suggestion
    """Return true if the shape/extent is **not equal** to the expected shape.
```